addons/survey/models/survey_question.py

```diff
@@ -707,6 +708,21 @@ def _check_question_not_empty(self):
             if not bool(label.question_id) != bool(label.matrix_question_id):
                 raise ValidationError(_("A label must be attached to only one question."))
 
+    @api.model
+    def _get_matching_answer_domain(self, row_id=False):
+        self.ensure_one()
+        if self.question_type == "matrix":
+            return [expression.AND([[('question_id', '=', self.question_id.id)], [('matrix_row_id', '=', row_id)],
```

Comment:
AND is the default operator, while it's useful for combining already created domains, here I would simply add the 3 tuples in a list, will make it easier to read.
(Same for the elif part).